### ISSUE TITLE
[Bug]: NewGRF animated tiles can be very resource-intensive

### ISSUE THREAD

**Issue by maaaaaaaaaad** [OpenTTD/OpenTTD#10192]
### Version of OpenTTD

13.0 - Beta 1 - OS 13.0.1

### Expected result

First of all, thank you to everyone who put work into this release and all the previous releases, and I apologize if this bug report doesn't belong here. 

I started a new game in 13.0 Beta -1, and noticed a very low frame rate even though the save game still has relatively few vehicles. I'm not sure if the issue is caused by the latest beta or is a result of possibly incompatible GRFs?

### Actual result

I expected a higher frame rate especially given that there are still very few vehicles in the save game. 

### Steps to reproduce

The save game is too large to upload but can be found [here](https://www.dropbox.com/s/a83gb9eqwdvz947/JR%2C%20Feb%2020th%2C%201974.sav.zip?dl=0)

**Comment by James103** [comment]
The attached save includes a NewGRF which is not available on the ingame content downloader.
Are you able to reproduce this without such NewGRFs?

**Comment by JGRennison** [comment]
That NewGRF is not essential to load the savegame and observe the issue.

In general you'd probably be better off with a smaller map size.

**Comment by maaaaaaaaaad** [comment]
True, but that would make it less fun :) I've played same-size scenarios before but encountered frame rate issues much later in the gameplay when it included thousands of vehicles. 

However, at that point, I used the Japanese TownSet (and not the RealHouses 32-bit version used here), which I suspect might be contributing to the lower frame rate?

**Comment by JGRennison** [comment]
Yes, the Real Houses set is causing the issue due to its use of animation callbacks for detecting neighbouring roads for building orientation. I'd suggest using a different house set in future games.

**Comment by maaaaaaaaaad** [comment]
Got it, that's a shame such a beautiful set, is there a way to possibly switch off the detecting mode?
